sql/sql_plugin.cc

```diff
@@ -3092,7 +3092,7 @@ static bool plugin_var_memalloc_session_update(THD *thd,
 
 void plugin_thdvar_safe_update(THD *thd, st_mysql_sys_var *var, char **dest, const char *value)
 {
-  DBUG_ASSERT(thd == current_thd);
+  DBUG_ASSERT(current_thd == NULL || thd == current_thd);
```

Comment:
This is upstream 8.0 bug, correct? Once the patch is final, can you open a bug report with them, mark LP as its downstream copy, and send the patch?